File: gamelogic/src/main/java/com/alta_v2/game/gamelogic/domain/npc/RepeatableActionProcessor.java
package com.alta_v2.game.gamelogic.domain.npc;

import com.alta_v2.common.Destroyable;
import com.alta_v2.game.gamelogic.data.npc.NpcModel;

/**
 * Describes the processor of repeatable action of NPC.
 */
public interface RepeatableActionProcessor extends Destroyable {

    /**
     * Adds given NPC into list. Each item of list tried to perform movement to random direction time to time.
     *
     * @param npc - the {@link NpcModel} to be used for repeatable movement.
     */
    void addToRandomMovement(NpcModel npc);

    /**
     * Starts processing of random movement.
     */
    void startAsync();
}

File: gamelogic/src/main/kotlin/com/alta_v2/game/gamelogic/domain/npc/RepeatableActionProcessor.kt
package com.alta_v2.game.gamelogic.domain.npc

import com.alta_v2.common.Destroyable
import com.alta_v2.game.gamelogic.data.npc.NpcModel

/**
 * Describes the processor of repeatable action of NPC.
 */
interface RepeatableActionProcessor : Destroyable {

    /**
     * Adds given NPC into list. Each item of list tried to perform movement to random direction time to time.
     *
     * @param npc - the [NpcModel] to be used for repeatable movement.
     */
    fun addToRandomMovement(npc: NpcModel)

    /**
     * Starts processing of random movement.
     */
    fun startAsync()

}